packages/client/src/utils/hooks/useWindowListener.ts

```diff
@@ -6,10 +6,14 @@ export const useWindowListener = <K extends keyof WindowEventMap>(
     deps: unknown[] = []
 ) => {
     useEffect(() => {
-        window.addEventListener(event, listener)
+        if (typeof window !== 'undefined') {
```

Comment:
Здесь достаточно выйти из эффекта при отсутствии window, чтоб убрать вложенность и не оборачивать каждое действие
```suggestion
        if (typeof window === 'undefined') {
            return undefined
        }
```